apps/src/code-studio/components/progress/LessonLockDataApi.js

```diff
@@ -0,0 +1,104 @@
+import {useMemo} from 'react';
+import _ from 'lodash';
+import {makeEnum} from '@cdo/apps/utils';
+import {useFetch} from '@cdo/apps/util/useFetch';
+
+export const LockStatus = makeEnum('Locked', 'Editable', 'ReadonlyAnswers');
+
+/**
+ * @typedef {Object} UserLockState
```

Comment:
Somewhat experimentally, I added more type annotations to this file.  This ended up showing really helpful annotations in the IDE.  Even though we're not using Typescript (yet), there can still be value in including more type annotations.